### PR TITLE
Add consistentVersions rule

### PR DESCRIPTION
**Pull request by thasner**
@styu

### CODE REVIEW

File: packages/rules/src/consistentVersions.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  const packageJsonPath = context.getPackageJsonPath();
+
+  const expectedPackageDependencyVersion = coerce(expectedPackageDependencyValue);
+  if (expectedPackageDependencyVersion == null) {
+    throw new Error(
+      `Malformed expected package dependency version defined in monorepolint configuration: ${dependencyPackageName} @ '${expectedPackageDependencyValue}'`
+    );
+  }
+
+  const actualPackageDependencyValue = packageJson.dependencies && packageJson.dependencies[dependencyPackageName];
```
Comment (thasner) [inline packages/rules/src/consistentVersions.ts:47]: prettier was not happy about `packageJson.dependencies?.[dependencyPackageName]`

Comment (styu) [reply packages/rules/src/consistentVersions.ts:47]: huh, why not?

Comment (thasner) [reply packages/rules/src/consistentVersions.ts:47]: Screen Shot 2020-06-05 at 2 45 39 PM
Screen Shot 2020-06-05 at 2 46 01 PM

Comment (styu) [reply packages/rules/src/consistentVersions.ts:47]: this is a typescript issue, this repo's version of typescript is too old:
https://github.com/monorepolint/monorepolint/blob/2d6068c00482889f16169cc1885d331fc5d66793/package.json#L61

**Comment by styu** [comment]
separately we should probably bump typescript at some point

**Pull request merged**